### ISSUE TITLE
Downloader cannot fetch modified date for Wine versions json

### ISSUE THREAD

**Issue by plata** [PhoenicisOrg/phoenicis#2144]
The `Downloader` always returns 0 when trying to determine the last modified date of https://phoenicis.playonlinux.com/index.php/wine?os=linux.
See: https://github.com/PhoenicisOrg/phoenicis/blob/6143a9fc3b38c40058f7dc72b6aa5da71b7852da/phoenicis-tools/src/main/java/org/phoenicis/tools/http/Downloader.java#L364
As a result, the file is downloaded every time even if this would not be required. I'm not sure if something is missing in the HTTP header or there is an implementation issue in `Downloader`.

**Comment by plata** [comment]
@qparis do you have an idea?

**Comment by qparis** [comment]
The URL https://phoenicis.playonlinux.com/index.php/wine?os=linux does not support last modified header. However, it should not be a problem because this file should be quick to download

**Comment by plata** [comment]
It is causing problems in the context of #1807. Of course it doesn't explain why the download does not start for so long in the first place but having the modified date would avoid the issue in many cases. Is there a reason why we cannot have the modified date for this URL?

**Comment by qparis** [comment]
The page is dynamically served.
I think this would just hide the issue for some people

**Comment by plata** [comment]
Normally the Wine versions are updated every two weeks. So this could sum up to a lot of useless downloads. Is there an alternative to using the last modified date?

**Comment by qparis** [comment]
They can be updated more often in fact (Patchs, etc...). 
We first need to understand if the slowliness issue is a server or a client problem.

**Comment by plata** [comment]
I don't know. What I can say: I experience the issue that the download does not start all the time. When I debug, `Downloader#isUpdateAvailable()` is executed immediately and can download the header/setup the connection right away.

**Comment by qparis** [comment]
That's right. However, most of the time, when this file is downloaded, it is just before the download of a way larger archive. The time required to download the json should be so small that we should not bother optimizing it. For example, in POLv4, the wineversion list has never been cached

**Comment by plata** [comment]
This is only partly true. If you use a Wine version which exists already on your local machine (typically the latest stable version), the json might be the only file that is downloaded at all (in case of a local app installation).
It is true that the download should not take long (~3.5MB) but it's often also not required. Maybe some users have a slow Internet connection. So if there's a rather easy way to detect when the file has been updated, I think we should implement it.

**Comment by qparis** [comment]
> This is only partly true. If you use a Wine version which exists already on your local machine (typically the latest stable version), the json might be the only file that is downloaded at all (in case of a local app installation).

Why? If you already have the required version, the list does not need to be downloaded, does it?

**Comment by plata** [comment]
Yes, if you do not specify the version but use e.g. the latest stable version. This is determined based on the list of available versions which requires that the versions json is fetched.
